registry/storage/driver/s3-aws/s3.go

```diff
@@ -1499,70 +1447,42 @@ func (w *writer) Write(p []byte) (int, error) {
 		}
 	}
 
-	var n int
-
-	defer func() { w.size += int64(n) }()
-
-	reader := bytes.NewReader(p)
-
-	for reader.Len() > 0 {
-		// NOTE(milosgajdos): we do some seemingly unsafe conversions
-		// from int64 to int in this for loop. These are fine as the
-		// offset returned from buffer.ReadFrom can only ever be
-		// maxChunkSize large which fits in to int. The reason why
-		// we return int64 is to play nice with Go interfaces where
-		// the buffer implements io.ReaderFrom interface.
+	n, _ := w.buf.Write(p)
```

Comment:
We should _probably_ make a note here saying that `error` is always `nil` when writing to `bytes.Buffer`.

Reply:
We are still missing comment here